tools/gen_customop_template.py

```diff
@@ -0,0 +1,98 @@
+'''
+Generates the C++ template code for a new custom operator.
+
+Usage: $ python tools\gen_customop_template.py onnx_model_filepath
+'''
+
+import argparse
+import onnx
+import pathlib
+import sys
+
+# Load and check ONNX model
+def load_onnx_model(model_path):
+    model = onnx.load(model_path)
+    onnx.checker.check_model(model)
+    return model
+
+# Get input and output type count from the ONNX model using protobuf information
+def get_io_count(model):
+    input_count = 0
+    output_count = 0
+    custom_op_node_exists = False
+
+    for node in model.graph.node:
+        # Find CustomOp node using domain
+        if node.domain == "ai.onnx.contrib" or node.domain == "com.microsoft.extensions":
+            custom_op_node_exists = True
+            input_count = len(node.input)
+            output_count = len(node.output)
+
+    if not custom_op_node_exists:
```

Comment:
what if there is more than one?

Reply:
From my conversation with @wenbingl I believe we are just handling one custom op node case as of now - this is just an initial wip, there is also more research to be done on shape inference to support the template more extensively.

Reply:
Generally, either you need to handle the case, or print a message to let the user know.

Reply:
Sounds good!